GliaWidgets/SecureConversations/SecureConversations.FileUploadList.swift

```diff
@@ -0,0 +1,965 @@
+            stackView.accessibilityElements?.append(fileUploadView.removeButton)
+        }
+
+        /// Remove fileUploadView` and `fileUploadView.removeButton` from
+        /// `accessibilityElements`.
+        func removeAccessibilityProperties(for fileUploadView: FileUploadView) {
+            stackView.accessibilityElements?.removeAll(
+                where: {
+                    guard let view = $0 as? UIView else { return false }
+                    return view === fileUploadView || view === fileUploadView.removeButton
+                }
+            )
+        }
+
+        /// Clear all `accessibilityElements`at once.
+        func removeAccessibilityPropertiesForAllUploadViews() {
+            stackView.accessibilityElements?.removeAll()
+        }
+
+        func renderProps() {
+            // Since we need to create/remove views based on the diff
+            // in collection of props, we need to have association
+            // between each element in collection of props and each
+            // corresponding view. This is achieved by having common
+            // id for props and view.
+
+            // Check for removal of views.
+            var toRemove = [FileUploadView.Props.Identifier]()
+            for id in cachedViews.ids where props.uploads[by: id] == nil {
+                toRemove.append(id)
+            }
+
+            for id in toRemove {
+                guard let view = cachedViews[by: id] else { continue }
+                removeUploadView(view)
+                cachedViews.remove(by: id)
+            }
+
+            // Check for addition of views.
+            for id in props.uploads.ids where cachedViews[by: id] == nil {
+                let uploadView = FileUploadView()
+                addUploadView(uploadView)
+                cachedViews.append(item: uploadView, identified: id)
+            }
+
+            // Assign props for corresponding views
+            for uploadProps in props.uploads {
+                cachedViews[by: uploadProps.id]?.props = uploadProps
+            }
+        }
+    }
+}
+
+extension SecureConversations.FileUploadListView {
+    class FileUploadView: UIView {
+        typealias FileUpload = Props
+
+        struct Props: Equatable, Identifiying {
+            typealias Identifier = String
+            let id: Identifier
+            let style: FileUploadStyle
+            let state: State
+            let removeTapped: Cmd
+        }
+
+        static let height: CGFloat = 60
+
+        private let contentView = UIView()
+        private let infoLabel = UILabel()
+        private let stateLabel = UILabel()
+        private let filePreviewView = FilePreviewView()
+        private let progressView = UIProgressView()
+        let removeButton = UIButton()
+
+        private let kContentInsets = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+        private let kRemoveButtonSize = CGSize(width: 30, height: 30)
```

Comment:
I understand that these are legacy, however, if you have chance, please drop `k` prefix for these variables.